First we will consider that the first bid is in range 160-200, and 2nd bid is in range 250-320.

In [1]:
def profit1(x,y):
    return ((x-160)*(320-x))+((y-x-50)*(320-y))

x,y=0,0
prof=float('-inf')
for i in range(160,201):
    for j in range(250,321):
        if(prof<profit1(i,j)):
            x,y= i,j
            prof=profit1(i,j)

print("Profit:", prof)
print(x,y)

Profit: 6025
200 285


this is a small profit, so i don't think going for this is good, we'll try putting x in range 200-250 now

In [2]:
def profit2(x,y):
    return ((40)*(320-x))+((y-250)*(320-y))

x,y=0,0
prof=float('-inf')
for i in range(200,251):
    for j in range(250,321):
        if(prof<profit2(i,j)):
            x,y= i,j
            prof=profit2(i,j)

print("Profit:", prof)
print(x,y)

Profit: 6025
200 285


this is the same. we'll put x in range 250-y now

In [3]:
def profit3(x,y):
    return ((x-210)*(320-x))+((y-x)*(320-y))

x,y=0,0
prof=float('-inf')
for j in range(250,321):
    for i in range(250,j+1):
        if(prof<profit3(i,j)):
            x,y= i,j
            prof=profit3(i,j)

print("Profit:", prof)
print(x,y)

Profit: 4025
250 285


this is even worse lmao.. now we will put x in 160-200 and y in 200-250

In [4]:
def profit4(x,y):
    return ((x-160)*(320-x))+((200-x)*(320-y))

x,y=0,0
prof=float('-inf')
for i in range(160,201):
    for j in range(200,321):
        if(prof<profit4(i,j)):
            x,y= i,j
            prof=profit4(i,j)

print("Profit:", prof)
print(x,y)

Profit: 5200
180 200


this seems decent, but not good enough... now well try both x and y in 200-250

In [5]:
def profit5(x,y):
    return 40*(320-x)

x,y=0,0
prof=float('-inf')
for j in range(200,251):
    for i in range(200,j+1):
        if(prof<profit5(i,j)):
            x,y= i,j
            prof=profit5(i,j)

print("Profit:", prof)
print(x,y)

Profit: 4800
200 200


why even bother... now x and y in range 160-200

In [6]:
def profit6(x,y):
    return ((x-160)*(320-x))+((y-x)*(320-y))

x,y=0,0
prof=float('-inf')
for j in range(160,200):
    for i in range(160,j+1):
        if(prof<profit6(i,j)):
            x,y= i,j
            prof=profit6(i,j)

print("Profit:", prof)
print(x,y)

Profit: 5099
179 199


somehow this is the best?! how tf??

So finally the best we found so far is bid1-200, bid2-285, with profit of around 6025 (*considering at each interval of 1 place, like 169-170, only1 turtle is there selling only 1 flipper) this also ignores the face that we are supposed to check the average bid. now if average bid is 285(considering all people put 285) we'll chek which one is the best for us now using probability they shaed on wiki.

In [14]:
def profit7(x,y):
    return ((x-160)*(320-x))+((y-x-50)*(320-y))*(min((35/(320-y))**3,1))
x,y=0,0
prof=float('-inf')
for i in range(160,201):
    for j in range(250,320):
        if(prof<profit7(i,j)):
            x,y= i,j
            prof=profit7(i,j)

print("Profit:", prof)
print(x,y)
print("Profit at 200,295: ", profit7(200,295))
print("Profit at 200,300: ", profit7(200,300))
print("Profit at 200,280: ", profit7(200,280))

Profit: 6025.0
200 285
Profit at 200,295:  5925
Profit at 200,300:  5800
Profit at 200,280:  5603.90625


we can see that making the 2nd bid as 295 or 300 will reduce our "expected" profit by 100 and 225 respectively, but that might also guarantee us the trade, because most people will go for the 285 bid. Going for below the threshold, we can see how rapodly the profit decreases. it decreased by 400 by just bidding 5 points below average. Also similar to last year, there might be a lot of bots increaseing(or decreasing, but mostly increasing) the average 2nd bid.